### DISCUSSION TITLE
Using statically compiled openssl libs in applications

### DISCUSSION THREAD

**Discussion by casidapulak** [openssl/openssl#21639] [Q&A]
Hello,
I have a c++ application App.exe which loads 2 dlls:

1. A.dll is loaded using implicit linking at the start of the program. It loads openssl dlls dynamically. All the openssl dlls(including fips.dll and legacy.dll) are present in the folder in which App.exe and A.dll are present. 
2. B.dll is loaded using explicit linking on demand basis using LoadLibrary. It is compiled using static lib files of openssl (B.dll has its own openssl settings like different providers etc. That is the reason it had to be done like this).

My question is if openssl search path needs to be set in B.dll to look for legacy.dll and fips.dll that are part of static lib compilation. Currently I have not done that and my application seems to work fine and B.dll seems to use legacy.dll and fips.dll which are part of dynamic lib compilation of openssl ( but if I check in dependency walker legacy.dll seems to have a dependency on the openssl dlls)

Thanks in advance!

**Comment by nhorman** [comment]
This appears to be a duplicate of #21680
